cpp/src/Slice/Grammar.y

```diff
@@ -116,6 +117,26 @@ slice_error(const char* s)
     }
 }
 
+namespace
+{
+    /// Resolves a `Type` named `name` from the current scope and returns it.
+    /// If no such type could be found, this returns `nullptr` instead.
+    /// @param name The (possibly scoped) name to resolve.
+    /// @param expectInterfaceType `true` is the type ends with '*' (indicated a proxy), false otherwise.
+    ///        This function will automatically emit errors for interfaces missing '*', or non-interfaces using '*'.
+    [[nodiscard]] TypePtr lookupTypeByName(const string& name, bool expectInterfaceType);
+
+    /// Resolves an `InterfaceDef` named `name` from the current scope and returns it.
+    /// If no such interface could be found, this returns `nullptr` instead.
+    /// @param name The (possibly scoped) name to resolve.
+    [[nodiscard]] InterfaceDefPtr lookupInterfaceByName(const string& name);
+
+    /// Checks if the provided integer is between '0' and 'int32_t::max' (inclusive).
```

Comment:
I would use interval notation. From your comment is it `[0, int32_t::max]` or `(0, int32_t::max]`

Reply:
Fair point, I guess this was ambiguous. I was meaning `[0, int32_t::max]` (fully inclusive).
I used the notation, and updated the comment to talk about the token, instead of the integer directly.